src/main/java/pro/trevor/tankgame/rule/impl/shared/PlayerRules.java

```diff
@@ -144,6 +144,7 @@ public static PlayerActionRule<Council> GetRuleCofferCostGrantLife(int cost)
                 if (t.isDead()) {
                     t.setDead(false);
                     t.setDurability(1);
+                    s.getCouncil().getCouncillors().remove(t.getPlayer());
```

Comment:
I was wondering if it would be easier for the council class to have API to add and remove players, so that the internal data structure isn't exposed. Or maybe something a bit more abstract, such as:

- AddCouncilor(string player);
- PromoteCouncilor(string player); // can throw errors if that player isn't a councilor
- RemoveMember(string player); // removes player from either senator or councilor list

Reply:
Basically, I think getCouncilors() and getSenators() is too exposed, and I would prefer these functions were not public.

Reply:
I can agree, but that would be a separate issue/PR

Reply:
Deferred